src/compiler/sys.ts

```diff
@@ -415,9 +465,10 @@ namespace ts {
                     // and is more efficient than `fs.watchFile` (ref: https://github.com/nodejs/node/pull/2649
                     // and https://github.com/Microsoft/TypeScript/issues/4643), therefore
                     // if the current node.js version is newer than 4, use `fs.watch` instead.
-                    const watchedFile = watchedFileSet.addFile(fileName, callback);
+                    const watchSet = isNode4OrLater() ? watchedFileSet : pollingWatchedFileSet;
+                    const watchedFile =  watchSet.addFile(fileName, callback);
                     return {
-                        close: () => watchedFileSet.removeFile(watchedFile)
+                        close: () => watchSet.removeFile(watchedFile)

```

Comment:
I tested before that if not supported then the `recursive: true` would simply be ignored. The recursive directory watcher is used with `tsconfig.json` without files array, so that adding/removing files in the background would be detected. The plan for this feature at the time was to support Node v4 or later only, so I didn't differentiate with the cases with older Node version.